### PR TITLE
5098 Branch 1: Groundwork for coming changes

### PR DESCRIPTION
**Pull request by Matilda-Clerke**
## PR description
Lay the groundwork for the full set of changes laid out in https://github.com/hyperledger/besu/pull/7389 with remaining changes to be added in additional, backwards compatible, PRs.

## Fixed Issue(s)

https://github.com/hyperledger/besu/issues/5098

### CODE REVIEW

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/response/RpcErrorType.java [hunk 1]
```diff
@@ -26,7 +26,77 @@ public enum RpcErrorType implements RpcMethodError {
   PARSE_ERROR(-32700, "Parse error"),
   INVALID_REQUEST(-32600, "Invalid Request"),
   METHOD_NOT_FOUND(-32601, "Method not found"),
+
   INVALID_PARAMS(-32602, "Invalid params"),
+  INVALID_ACCOUNTS_PARAMS(-32602, "Invalid accounts params"),
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/response/RpcErrorType.java:31]:
```suggestion
  INVALID_ACCOUNT_PARAMS(-32602, "Invalid account params"),
```

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/response/RpcErrorType.java [hunk 2]
```diff
@@ -26,7 +26,77 @@ public enum RpcErrorType implements RpcMethodError {
...
+  INVALID_BLOCK_NUMBER_PARAMS(-32602, "Invalid block number params"),
+  INVALID_CALL_PARAMS(-32602, "Invalid call params"),
+  INVALID_CONSOLIDATION_REQUEST_PARAMS(-32602, "Invalid consolidation request params"),
+  INVALID_CREATE_PRIVACY_GROUP_PARAMS(-32602, "Invalid create privacy group params"),
+  INVALID_DATA_PARAMS(-32602, "Invalid data params"),
+  INVALID_DEPOSIT_REQUEST_PARAMS(-32602, "Invalid deposit request"),
+  INVALID_ENGINE_EXCHANGE_TRANSITION_CONFIGURATION_PARAMS(
+      -32602, "Invalid engine exchange transition configuration params"),
+  INVALID_ENGINE_FORKCHOICE_UPDATED_PARAMS(-32602, "Invalid engine forkchoice updated params"),
+  INVALID_ENGINE_PAYLOAD_ATTRIBUTES_PARAMETER(-32602, "Invalid engine payload attributes params"),
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/response/RpcErrorType.java:50]:
```suggestion
  INVALID_ENGINE_PAYLOAD_ATTRIBUTES_PARAMS(-32602, "Invalid engine payload attributes parameter"),
```

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/response/RpcErrorType.java [hunk 3]
```diff
@@ -26,7 +26,77 @@ public enum RpcErrorType implements RpcMethodError {
...
+  INVALID_TRACE_TYPE_PARAMS(-32602, "Invalid trace type params"),
+  INVALID_TRANSACTION_PARAMS(-32602, "Invalid transaction params (missing or incorrect)"),
+  INVALID_TRANSACTION_HASH_PARAMS(-32602, "Invalid transaction hash params"),
+  INVALID_TRANSACTION_ID_PARAMS(-32602, "Invalid transaction id params"),
+  INVALID_TRANSACTION_INDEX_PARAMS(-32602, "Invalid transaction index params"),
+  INVALID_TRANSACTION_LIMIT_PARAMS(-32602, "Invalid transaction limit params"),
+  INVALID_TRANSACTION_TRACE_PARAMS(-32602, "Invalid transaction trace params"),
+  INVALID_VERSIONED_HASH_PARAMS(-32602, "Invalid versioned hash params"),
+  INVALID_VOTE_TYPE_PARAMS(-32602, "Invalid vote type params"),
+  INVALID_WITHDRAWALS_PARAMS(-32602, "Invalid  withdrawals"),
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/response/RpcErrorType.java:98]:
```suggestion
  INVALID_WITHDRAWALS_PARAMS(-32602, "Invalid withdrawals"),
```

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/response/RpcErrorType.java [hunk 4]
```diff
@@ -26,7 +26,77 @@ public enum RpcErrorType implements RpcMethodError {
...
+  INVALID_PAYLOAD_ID_PARAMS(-32602, "Invalid payload id params"),
+  INVALID_PENDING_TRANSACTIONS_PARAMS(-32602, "Invalid pending transactions params"),
+  INVAlID_PLUGIN_NAME_PARAMS(-32602, "Invalid plug in name params"),
+  INVALID_POSITION_PARAMS(-32602, "Invalid position params"),
+  INVALID_POW_HASH_PARAMS(-32602, "Invalid pow hash params"),
+  INVALID_PRIVACY_GROUP_PARAMS(-32602, "Invalid privacy group params"),
+  INVALID_PRIVATE_FROM_PARAMS(-32602, "Invalid private from params"),
+  INVALID_PRIVATE_FOR_PARAMS(-32602, "Invalid private for params"),
+  INVALID_REMOTE_CAPABILITIES_PARAMS(-32602, "Invalid remote capabilities params"),
+  INVALID_REWARD_PERCENTILES_PARAMETER(-32602, "Invalid reward percentiles params"),
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/response/RpcErrorType.java:81]:
```suggestion
  INVALID_REWARD_PERCENTILES_PARAMS(-32602, "Invalid reward percentiles params"),
```

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/response/RpcErrorType.java [hunk 5]
```diff
@@ -26,7 +26,77 @@ public enum RpcErrorType implements RpcMethodError {
...
+  INVALID_CALL_PARAMS(-32602, "Invalid call params"),
+  INVALID_CONSOLIDATION_REQUEST_PARAMS(-32602, "Invalid consolidation request params"),
+  INVALID_CREATE_PRIVACY_GROUP_PARAMS(-32602, "Invalid create privacy group params"),
+  INVALID_DATA_PARAMS(-32602, "Invalid data params"),
+  INVALID_DEPOSIT_REQUEST_PARAMS(-32602, "Invalid deposit request"),
+  INVALID_ENGINE_EXCHANGE_TRANSITION_CONFIGURATION_PARAMS(
+      -32602, "Invalid engine exchange transition configuration params"),
+  INVALID_ENGINE_FORKCHOICE_UPDATED_PARAMS(-32602, "Invalid engine forkchoice updated params"),
+  INVALID_ENGINE_PAYLOAD_ATTRIBUTES_PARAMETER(-32602, "Invalid engine payload attributes params"),
+  INVALID_ENGINE_PAYLOAD_PARAMETER(-32602, "Invalid engine payload params"),
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/response/RpcErrorType.java:51]:
```suggestion
  INVALID_ENGINE_PAYLOAD_PARAMS(-32602, "Invalid engine payload parameter"),
```

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/response/RpcErrorType.java [hunk 6]
```diff
@@ -26,7 +26,77 @@ public enum RpcErrorType implements RpcMethodError {
...
+  INVALID_ENGINE_PREPARE_PAYLOAD_PARAMETER(-32602, "Invalid engine prepare payload parameter"),
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/response/RpcErrorType.java:52]:
```suggestion
  INVALID_ENGINE_PREPARE_PAYLOAD_PARAMS(-32602, "Invalid engine prepare payload parameter"),
```

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/response/RpcErrorType.java [hunk 7]
```diff
@@ -26,7 +26,77 @@ public enum RpcErrorType implements RpcMethodError {
...
+  INVALID_PRIVACY_GROUP_PARAMS(-32602, "Invalid privacy group params"),
+  INVALID_PRIVATE_FROM_PARAMS(-32602, "Invalid private from params"),
+  INVALID_PRIVATE_FOR_PARAMS(-32602, "Invalid private for params"),
+  INVALID_REMOTE_CAPABILITIES_PARAMS(-32602, "Invalid remote capabilities params"),
+  INVALID_REWARD_PERCENTILES_PARAMETER(-32602, "Invalid reward percentiles params"),
+  INVALID_SEALER_ID_PARAMS(-32602, "Invalid sealer ID params"),
+  INVALID_STORAGE_KEYS_PARAMS(-32602, "Invalid storage keys params"),
+  INVALID_SUBSCRIPTION_PARAMS(-32602, "Invalid subscription params"),
+  INVALID_TARGET_GAS_LIMIT_PARAMS(-32602, "Invalid target gas limit params"),
+  INVALID_TIMESTAMP_PARAMETER(-32602, "Invalid timestamp params"),
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/response/RpcErrorType.java:86]:
```suggestion
  INVALID_TIMESTAMP_PARAMS(-32602, "Invalid timestamp parameter"),
```

**Review by macfarla — APPROVED** [review]
few nits on naming consistency

also wondering whether -32602 should be a constant since it's repeated so many times?  If it would help readability - INVALID_PARAMS_ERROR_CODE ?

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/response/RpcErrorType.java (hunk 7 above)
Comment (Matilda-Clerke) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/response/RpcErrorType.java]: Good calls, updated

**Review by macfarla — APPROVED** [review]

**Review by jframe — APPROVED** [review]

**Pull request merged**